pytensor/tensor/shape.py

```diff
@@ -972,10 +972,7 @@ def make_node(self, x):
         x = as_tensor_variable(x)
         if x.type.ndim <= max(self.axes):
             raise ValueError("Trying to unbroadcast of non-existent dimension")
-        shape = [
-            None if (sh == 1 and i in self.axes) else sh
-            for i, sh in enumerate(x.type.shape)
-        ]
+        shape = [None if i in self.axes else sh for i, sh in enumerate(x.type.shape)]
```

Comment:
This is losing more information than before. IIRC it's fine to unbroadcast an axis that's already known to be non-broadcastable.

The perform is actually just a pass through, no checks are done